lib/matplotlib/tests/test_quiver.py

```diff
@@ -333,3 +334,93 @@ def test_quiver_setuvc_numbers():
     q.set_UVC(0, 1)
+
+
+def draw_quiverkey_zorder_argument(fig, zorder=None):
+    """Draw Quiver and QuiverKey using zorder argument"""
+    x = np.arange(1, 6, 1)
+    y = np.arange(1, 6, 1)
+    X, Y = np.meshgrid(x, y)
+    U, V = 2, 2
+
+    ax = fig.subplots()
+    q = ax.quiver(X, Y, U, V, pivot='middle')
+    ax.set_xlim(0.5, 5.5)
+    ax.set_ylim(0.5, 5.5)
+    if zorder is None:
+        ax.quiverkey(q, 4, 4, 25, coordinates='data',
+                     label='U', color='blue')
+        ax.quiverkey(q, 5.5, 2, 20, coordinates='data',
+                     label='V', color='blue', angle=90)
+    else:
+        ax.quiverkey(q, 4, 4, 25, coordinates='data',
+                     label='U', color='blue', zorder=zorder)
+        ax.quiverkey(q, 5.5, 2, 20, coordinates='data',
+                     label='V', color='blue', angle=90, zorder=zorder)
+
+
+def draw_quiverkey_setzorder(fig, zorder=None):
+    """Draw Quiver and QuiverKey using set_zorder"""
+    x = np.arange(1, 6, 1)
+    y = np.arange(1, 6, 1)
+    X, Y = np.meshgrid(x, y)
+    U, V = 2, 2
+
+    ax = fig.subplots()
+    q = ax.quiver(X, Y, U, V, pivot='middle')
+    ax.set_xlim(0.5, 5.5)
+    ax.set_ylim(0.5, 5.5)
+    qk1 = ax.quiverkey(q, 4, 4, 25, coordinates='data',
+                       label='U', color='blue')
+    qk2 = ax.quiverkey(q, 5.5, 2, 20, coordinates='data',
+                       label='V', color='blue', angle=90)
+    if zorder is not None:
+        qk1.set_zorder(zorder)
+        qk2.set_zorder(zorder)
+
+
+@check_figures_equal(extensions=['png'])
+def test_quiverkey_zorder_default(fig_test, fig_ref):
+    """Check QuiverKey zorder argument"""
+    draw_quiverkey_zorder_argument(fig_test)
+    draw_quiverkey_setzorder(fig_ref)
+
+
+@check_figures_equal(extensions=['png'])
+def test_quiverkey_zorder_zero(fig_test, fig_ref):
+    """
+    Check QuiverKey zorder argument
+    zorder=0 means quiverkey is under quiver.
+    """
+    draw_quiverkey_zorder_argument(fig_test, zorder=0)
+    draw_quiverkey_setzorder(fig_ref, zorder=0)
+
+
+@check_figures_equal(extensions=['png'])
+def test_quiverkey_zorder_two(fig_test, fig_ref):
+    """
+    Check QuiverKey zorder argument
+    zorder=2 means quiverkey is same as default.
+    """
+    draw_quiverkey_zorder_argument(fig_test, zorder=2)
+    draw_quiverkey_setzorder(fig_ref, zorder=2)
+
+
+@check_figures_equal(extensions=['png'])
+def test_quiverkey_zorder_five(fig_test, fig_ref):
+    """
+    Check QuiverKey zorder argument
+    zorder=5 means quiverkey is over ticks.
+    """
+    draw_quiverkey_zorder_argument(fig_test, zorder=5)
+    draw_quiverkey_setzorder(fig_ref, zorder=5)
+
+
+@check_figures_equal(extensions=['png'])
+def test_quiverkey_zorder_None(fig_test, fig_ref):
+    """
+    Check QuiverKey zorder argument
+    zorder=2 means quiverkey is default.
+    """
+    draw_quiverkey_zorder_argument(fig_test, zorder=None)
+    draw_quiverkey_setzorder(fig_ref, zorder=None)
```

Comment:
These tests can be combined with `pytest.mark.parametrize`:
```suggestion
@pytest.mark.parametrize('zorder', [0, 2, 5, None])
@check_figures_equal(extensions=['png'])
def test_quiverkey_zorder(fig_test, fig_ref, zorder):
    draw_quiverkey_zorder_argument(fig_test, zorder=zorder)
    draw_quiverkey_setzorder(fig_ref, zorder=zorder)
```
Note, I removed `test_quiverkey_zorder_default`, because it is just testing the default of the `draw_*` helpers, not `quiverkey`, and it's functionally equivalent to passing `None`.

And then, since they're only called once, you probably should inline the `draw_*` helper functions.